### PR TITLE
Ensure only one TeacherProfile per user

### PR DESCRIPTION
**Pull request by ethanmills**
### Context
Each user should have at most one teacher profile. Add a database constraint to ensure that.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -585,8 +585,11 @@
     t.string "administrative_district_code"
     t.string "administrative_district_name"
     t.string "slug"
+    t.boolean "section_41_approved", default: false, null: false
```
Comment (asmega) [inline db/schema.rb:588]: are these suppose to be here? as one would expect the migration for these to be present in this pr

Comment (ethanmills) [reply db/schema.rb:588]: uh oh

File: spec/cypress/integration/nominations/NominateInductionCoordinator.feature [hunk 2]
```diff
@@ -56,7 +56,7 @@ Feature: Nominate induction tutor
     And the page should be accessible
     And percy should be sent snapshot called "Start nominations name different"
 
-    Given user was created as "teacher" with email "different-user-type@example.com"
+    Given user was created with email "different-user-type@example.com"
```
Comment (kicferk1) [inline spec/cypress/integration/nominations/NominateInductionCoordinator.feature:59]: Out of curiosity, did this start to fail here, or is it a cheeky fix of an incorrect test?

**Review by kicferk1 — APPROVED** [review]

File: spec/cypress/integration/nominations/NominateInductionCoordinator.feature (hunk 2 above)
Comment (ethanmills) [reply spec/cypress/integration/nominations/NominateInductionCoordinator.feature:59]: This started to fail here. This line created a teacher profile for the user, then the next line creates another

**Pull request merged**